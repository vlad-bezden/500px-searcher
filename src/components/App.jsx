import React from 'react'

import Header from './Header'
import Search from './Search'

class App extends React.Component {
  render() {
    return (
      <div className='container'>
        <div className='row'>
          <Header />
        </div>
        <div className='row'>
          <Search />
        </div>
      </div>
    )
  }
}

export default App