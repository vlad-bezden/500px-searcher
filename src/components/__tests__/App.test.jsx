import React from 'react'
import { shallow } from 'enzyme'

import App from '../App'

test('App renders correctly', () => {
  const component = shallow(
    <App />
  )

  const tree = component.getNode()

  expect(tree).toMatchSnapshot()
})

test('App should set modalVisible to false when user click on hide button', () => {
  const component = shallow(
    <App />
  )
  component.setState({
    modalVisible: true
  })

  component.find('PhotoModal').prop('onHide')()

  expect(component.state('modalVisible')).toBe(false)
})