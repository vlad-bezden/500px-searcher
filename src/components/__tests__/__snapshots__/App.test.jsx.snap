// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <Header />
  </div>
  <div
    className="row"
  >
    <Search
      onShowPhoto={[Function]}
    />
  </div>
  <PhotoModal
    onHide={[Function]}
    photo={null}
    show={false}
  />
</div>
`;
